### PR TITLE
web: Convert demo to typescript and copy the new extension-player styling over

### PR DESCRIPTION
**Pull request by Dinnerbone**
The demo now has the improvements from #13473, and uses the actual API through typescript

### CODE REVIEW

**Review by danielhjacobs — APPROVED** [review]

**Review by danielhjacobs — COMMENTED** [review]
This code is needed to make the "Select File" button work:

https://github.com/ruffle-rs/ruffle/blob/cad162121b18b29718ef58643430c2123badcc32/web/packages/extension/src/player.ts#L272-L276

**Comment by danielhjacobs** [comment]
The location of the Ruffle logo is not ideal on small screens (note: this is this PR with the URL input parameter set to true to make the screen as crowded as possible).

image

File: web/packages/demo/src/index.ts [hunk 1]
```diff
@@ -1,43 +1,72 @@
...
+let player: RufflePlayer | null;
 
-const main = document.getElementById("main");
-const overlay = document.getElementById("overlay");
-const authorContainer = document.getElementById("author-container");
-const author = document.getElementById("author");
+const main = document.getElementById("player-container")!;
+const overlay = document.getElementById("overlay")!;
+const authorContainer = document.getElementById("author-container")!;
+const author = <HTMLLinkElement>document.getElementById("author");
```
[the remark this answers is not in the document]
Comment (Dinnerbone) [reply web/packages/demo/src/index.ts:33]: I'll fix it here and add a lint in a future PR to enforce `as`

File: web/packages/demo/src/index.ts [hunk 2]
```diff
@@ -1,43 +1,72 @@
...
+const ruffle = (window.RufflePlayer as PublicAPI).newest()!;
 
-let player;
+let player: RufflePlayer | null;
 
-const main = document.getElementById("main");
-const overlay = document.getElementById("overlay");
-const authorContainer = document.getElementById("author-container");
-const author = document.getElementById("author");
+const main = document.getElementById("player-container")!;
```
[the remark this answers is not in the document]
Comment (Dinnerbone) [reply web/packages/demo/src/index.ts:30]: Good call. I renamed it back just to help work with the diff, but I've renamed it again now

**Comment by Dinnerbone** [comment]
Made it hide the logo when the screen is <=600px wide

**Pull request merged**